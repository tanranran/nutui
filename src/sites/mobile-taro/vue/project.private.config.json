{
  "setting": {
    "compileHotReLoad": true
  },
  "description": "项目私有配置文件。此文件中的内容将覆盖 project.config.json 中的相同字段。项目的改动优先同步到此文件中。详见文档：https://developers.weixin.qq.com/miniprogram/dev/devtools/projectconfig.html",
  "condition": {
    "miniprogram": {
      "list": [
        {
          "name": "exhibition/pages/imagepreview/index",
          "pathName": "exhibition/pages/imagepreview/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "exhibition/pages/popover/index",
          "pathName": "exhibition/pages/popover/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "exhibition/pages/countdown/index",
          "pathName": "exhibition/pages/countdown/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/collapse/index",
          "pathName": "feedback/pages/collapse/index",
          "query": "",
          "scene": null
        },
        {
          "name": "exhibition/pages/ellipsis/index",
          "pathName": "exhibition/pages/ellipsis/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "exhibition/pages/popover/index",
          "pathName": "exhibition/pages/popover/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "dentry/pages/textarea/index",
          "pathName": "dentry/pages/textarea/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "dentry/pages/picker/index",
          "pathName": "dentry/pages/picker/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        }
      ]
    }
  },
  "projectname": "vue",
  "libVersion": "2.27.1"
}