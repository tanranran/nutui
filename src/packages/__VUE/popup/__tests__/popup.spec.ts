import { mount, config } from '@vue/test-utils';
import { nextTick } from 'vue';
import PopUp from '../index.vue';
import NutOverlay from '../../overlay/index.vue';
import NutIcon from '../../icon/index.vue';

beforeAll(() => {
  config.global.components = {
    NutOverlay,
    NutIcon
  };
});

afterAll(() => {
  config.global.components = {};
});

function sleep(delay = 0): Promise<void> {
  return new Promise((resolve) => {
    setTimeout(resolve, delay);
  });
}

test('should change z-index when using z-index prop', async () => {
  const wrapper = mount(PopUp, {
    props: {
      visible: true,
      zIndex: 99,
      teleportDisable: false
    }
  });
  await nextTick();
  const pop: any = wrapper.find('.nut-popup');
  expect(pop.element.style.zIndex).toEqual('99');
});

test('should change animation duration when using duration prop', () => {
  const wrapper = mount(PopUp, {
    props: {
      visible: true,
      teleportDisable: false,
      duration: 12
    }
  });

  const overlay: any = wrapper.find('.nut-overlay');
  expect(overlay.element.style.transitionDuration).toEqual('12s');
});

test('should lock scroll when showed', async () => {
  const wrapper = mount(PopUp, {
    visible: false,
    teleportDisable: false
  });

  await wrapper.setProps({ visible: true });
  expect(document.body.classList.contains('nut-overflow-hidden')).toBeFalsy();
});

test('should not render overlay when overlay prop is false', () => {
  const wrapper = mount(PopUp, {
    props: {
      visible: true,
      teleportDisable: false,
      overlay: false
    }
  });
  expect(wrapper.find('.nut-overlay').exists()).toBeFalsy();
});

test('prop close-on-click-overlay test', async () => {
  const wrapper = mount(PopUp, {
    props: {
      visible: true,
      teleportDisable: false,
      closeOnClickOverlay: false
    }
  });
  await wrapper.trigger('click');
  const overlay: any = wrapper.find('.nut-overlay');
  expect(overlay.element.style.display).toEqual('');
});

test('pop from top', async () => {
  const wrapper = mount(PopUp, {
    props: {
      visible: true,
      teleportDisable: false,
      position: 'top'
    }
  });
  await nextTick();
  const pop: any = wrapper.find('.popup-top');
  expect(pop.exists()).toBeTruthy();
});

test('pop from bottom', async () => {
  const wrapper = mount(PopUp, {
    props: {
      visible: true,
      teleportDisable: false,
      position: 'bottom'
    }
  });
  await nextTick();
  const pop: any = wrapper.find('.popup-bottom');
  expect(pop.exists()).toBeTruthy();
});

test('pop from left', async () => {
  const wrapper = mount(PopUp, {
    props: {
      visible: true,
      teleportDisable: false,
      position: 'left'
    }
  });
  await nextTick();
  const pop: any = wrapper.find('.popup-left');
  expect(pop.exists()).toBeTruthy();
});

test('pop from right', async () => {
  const wrapper = mount(PopUp, {
    props: {
      visible: true,
      teleportDisable: false,
      position: 'right'
    }
  });
  await nextTick();
  const pop: any = wrapper.find('.popup-right');
  expect(pop.exists()).toBeTruthy();
});

test('should render close icon when using closeable prop', async () => {
  const wrapper = mount(PopUp, {
    props: {
      visible: true,
      teleportDisable: false,
      closeable: true
    }
  });
  await nextTick();
  const closeIcon = wrapper.find('.nut-popup__close-icon');
  expect(closeIcon.exists()).toBeTruthy();
});

test('should render correct close icon when using close-icon prop', () => {
  const wrapper = mount(PopUp, {
    props: {
      visible: true,
      teleportDisable: false,
      closeable: true,
      closeIcon: 'success'
    }
  });

  expect(wrapper.find('.nut-popup__close-icon').html()).toMatchSnapshot();
});

test('should have "nut-popup--round" class when setting the round prop', () => {
  const wrapper = mount(PopUp, {
    props: {
      visible: true,
      teleportDisable: false,
      round: true
    }
  });

  expect(wrapper.find('.round').exists()).toBeTruthy();
});

test('event click pop test', async () => {
  const wrapper = mount(PopUp, {
    props: {
      visible: true,
      teleportDisable: false,
      closeOnClickOverlay: true
    }
  });
  await nextTick();
  const popup: any = wrapper.find('.nut-popup');
  await popup.trigger('click');
  expect(wrapper.emitted('click-pop')).toBeTruthy();
});

test('event click-close-icon test', async () => {
  const wrapper = mount(PopUp, {
    props: {
      visible: true,
      teleportDisable: false,
      closeable: true
    }
  });
  await wrapper.find('.nut-popup__close-icon').trigger('click');
  expect(wrapper.emitted('click-close-icon')).toBeTruthy();
});

test('should emit open event when prop visible is set to true', async () => {
  const wrapper = mount(PopUp, {
    props: {
      visible: false,
      teleportDisable: false
    }
  });

  await wrapper.setProps({ visible: true });
  expect(wrapper.emitted('open')).toBeTruthy();
});

test('event close test', async () => {
  const wrapper = mount(PopUp, {
    props: {
      visible: true,
      teleportDisable: false
    }
  });
  await wrapper.find('.nut-overlay').trigger('click');
  await sleep(2000);
  expect(wrapper.emitted('click-overlay')).toBeTruthy();
});

test('event click-overlay test', async () => {
  const wrapper = mount(PopUp, {
    props: {
      visible: true,
      teleportDisable: false
    }
  });

  const overlay: any = wrapper.find('.nut-overlay');
  await overlay.trigger('click');
  expect(wrapper.emitted('click-overlay')).toBeTruthy();
});
