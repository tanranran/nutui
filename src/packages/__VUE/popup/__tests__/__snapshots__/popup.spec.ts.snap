// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correct close icon when using close-icon prop 1`] = `"<view class=\\"nut-popup__close-icon nut-popup__close-icon--top-right\\"><i class=\\"nutui-iconfont nut-icon nut-icon-success\\" style=\\"font-size: 12px; width: 12px; height: 12px;\\" src=\\"\\"></i></view>"`;
